src/pages/EditorPage/EditorPanel.jsx

```diff
@@ -506,28 +510,53 @@ const EditorPanel = ({
     <Section wrap id="editor-entity-details-container">
       {!noSelection && (
         <>
-          <EntityDetailsHeader
-            values={nodeIds.map((id) => nodes[id]?.data)}
-            tools={
-              <>
-                <Button
-                  icon="replay"
-                  onClick={handleRevert}
-                  disabled={noSelection}
-                  data-tooltip="Clear changes"
-                />
-                <Button
-                  icon="delete"
-                  onClick={() => onDelete(nodes)}
-                  disabled={noSelection}
-                  data-tooltip="Delete"
-                />
-                <Link to={`/projects/${projectName}/browser`}>
-                  <Button icon="visibility" disabled={noSelection} data-tooltip="View in Browser" />
-                </Link>
-              </>
-            }
-          />
+        <Container>
+          <EntityThumbnailUploader
+            isCompact
+            entities={isFetchingEntitiesDetails ? entitiesToQuery : entityDetailsData}
+            projectName={projectName}
+            entityType={entityType}
+            onUploaded={operations => {
+              const firstOperation = operations[0]
```

Comment:
Only one thumbnail gets updates, which makes sense in this code. Use a forEach loop to update all.

![image](https://github.com/user-attachments/assets/909ecfe8-425c-48fb-98db-1866066326b5)